src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4317,14 +4317,30 @@ public override bool PreVisit(CallNode node)
                         _txb.AddVolatileVariables(args[i], volatileVariables);
                     }
 
+                    var isIdentifier =
+                        args[i] is FirstNameNode &&
+                        _features.HasFlag(Features.SupportColumnNamesAsIdentifiers) &&
+                        maybeFunc.IsIdentifierParam(i);
+
                     // Use the new scope only for lambda args.
                     _currentScope = (maybeFunc.IsLambdaParam(i) && scopeInfo.AppliesToArgument(i)) ? scopeNew : scopeNew.Parent;
-                    args[i].Accept(this);
 
-                    _txb.AddVolatileVariables(node, _txb.GetVolatileVariables(args[i]));
+                    if (!isIdentifier)
+                    {
+                        args[i].Accept(this);
+                        _txb.AddVolatileVariables(node, _txb.GetVolatileVariables(args[i]));
+                        argTypes[i] = _txb.GetType(args[i]);
+                    }
+                    else
+                    {
+                        // This is an identifier, no associated type
+                        argTypes[i] = null;
```

Comment:
I do not think we should be doing this.

Reply:
Per teams chat today, this should be DType.Invalid.